tests/test_generator.py

```diff
@@ -170,6 +174,67 @@ def test_derivation_body() -> None:
     assert_body(generator)
 
 
+@pytest.mark.parametrize(
+    "left,right",
+    [
+        (expr.Variable("Value"), expr.Aggregate(expr.Number(1), expr.Number(2))),
+        (expr.Aggregate(expr.Number(1), expr.Number(2)), expr.Variable("Value")),
+    ],
+)
+@pytest.mark.parametrize("relation", [expr.Equal, expr.NotEqual])
+def test_substitution_relation_aggregate(
+    relation: Callable[[expr.Expr, expr.Expr], expr.Relation], left: expr.Expr, right: expr.Expr
+) -> None:
+    equal_call = expr.Call(
+        "Equal",
+        [
+            expr.Variable("Ctx"),
+            expr.Variable("F_Value"),
+            expr.Aggregate(
+                expr.Val(expr.Variable("Types.Byte"), expr.Number(1)),
+                expr.Val(expr.Variable("Types.Byte"), expr.Number(2)),
+            ),
+        ],
+    )
+
+    assert_equal(
+        relation(left, right).substituted(common.substitution(TLV_MESSAGE)),
+        equal_call if relation == expr.Equal else expr.Not(equal_call),
+    )
+
+
+@pytest.mark.parametrize(
+    "expressions,expected",
+    [
+        (
+            (expr.Variable("Length"), expr.Number(1)),
+            (expr.Call("Get_Length", [expr.Variable("Ctx")]), expr.Number(1)),
+        ),
+        (
+            (expr.Number(1), expr.Variable("Length")),
+            (expr.Number(1), expr.Call("Get_Length", [expr.Variable("Ctx")])),
+        ),
+        ((expr.Number(1), expr.Variable("Unknown")), (expr.Number(1), expr.Variable("Unknown"))),
+    ],
+)
+@pytest.mark.parametrize("relation", [expr.Less, expr.Equal, expr.Greater, expr.NotEqual])
```

Comment:
For sake of completeness, do we also want to test `LessEqual` and `GreaterEqual`?

Reply:
Yes, I will add them.

Reply:
Fixed in 9bbb3399c51e2deba2755a90d074ed6f7e8daa74.